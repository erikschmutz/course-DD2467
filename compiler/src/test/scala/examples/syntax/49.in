type color = RED | GREEN | BLUE | RGB of int * int * int;;
