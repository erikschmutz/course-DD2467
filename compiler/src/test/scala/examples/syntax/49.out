Program(
    List(
        TypeDeclaration(
            color,
            TypeConstraints(
                List(
                    TypeConstraint(RED,None),
                    TypeConstraint(GREEN,None),
                    TypeConstraint(BLUE,None),
                    TypeConstraint(
                        RGB,
                        Some(
                            TypeTuple(
                                List(
                                    TypePrimitive(int),
                                    TypePrimitive(int),
                                    TypePrimitive(int)
                                )
                            )
                        )
                    )
                )
            ),
            List()
        )
    )
)